index.test.js

```diff
@@ -57,3 +59,23 @@ describe("handles quotes around inline code", async () => {
       "`);
   });
 });
+
+describe("should ignore parent nodes", () => {
+  const mdxCompiler = remark().use(remarkMdx).use(remarkSmartypants);
+  const process = mdxCompiler.process.bind(mdxCompiler);
+
+  it("<style>", async () => {
+    const file = await process(`<style>html:after \\{ content: '""' }</style>`);
+    expect(file.value).toMatchInlineSnapshot(`
+      "<style>html:after \\{ content: '""' }</style>
+      "
+    `);
+  });
+  it("<script>", async () => {
+    const file = await process('<script>console.log("foo")</script>');
+    expect(file.value).toMatchInlineSnapshot(`
+      "<script>console.log("foo")</script>
+      "
+    `);
+  });
+});
```

Comment:
In this case the tests might be more readable if you saved the content in the variable. For example:

```ts
const markdownContent = '<script>console.log("foo")</script>'
const file = await process(markdownContent)
expect(file.value).toBe(markdownContent)
```

That way it's immediately obvious that nothing should have been changed.